### PR TITLE
Update the sarif schema file

### PR DESCRIPTION
**Pull request by aeisenberg**
The version we were using is quite old. Copied the latest from https://github.com/oasis-tcs/sarif-spec/blob/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json

I do not think the sarif spec will be changing any more without an explicit version update, so this is fine for now.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by aeisenberg** [comment]
Oh...thanks interesting. The failing check is also failing on main. https://github.com/github/codeql-action/actions/runs/4865932424/jobs/8676880004?pr=1668

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## [UNRELEASED]
 
 - You can now configure CodeQL within your code scanning workflow by passing a `config` input to the `init` Action. See [Using a custom configuration file](https://aka.ms/code-scanning-docs/config-file) for more information about configuring code scanning. [#1590](https://github.com/github/codeql-action/pull/1590)
+- Updated the SARIF JSON schema file to the latest from [oasis-tcs/sarif-spec](https://github.com/oasis-tcs/sarif-spec/blob/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json). [#1668](https://github.com/github/codeql-action/pull/1668)
```
Comment (adityasharad) [inline CHANGELOG.md:6]:
```suggestion
- Updated the SARIF 2.1.0 JSON schema file to the latest from [oasis-tcs/sarif-spec](https://github.com/oasis-tcs/sarif-spec/blob/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json). [#1668](https://github.com/github/codeql-action/pull/1668)
```

**Review by adityasharad — COMMENTED** [review]
LGTM, couple of minor suggestions. Thanks for catching this.

File: src/sarif-schema-2.1.0.json [hunk 2]
```diff
@@ -2339,8 +2368,9 @@
           "description": "The language of the messages emitted into the log file during this run (expressed as an ISO 639-1 two-letter lowercase culture code) and an optional region (expressed as an ISO 3166-1 two-letter uppercase subculture code associated with a country or region). The casing is recommended but not required (in order for this data to conform to RFC5646).",
           "type": "string",
           "default": "en-US",
-          "pattern": "^[a-zA-Z]{2}(-[a-zA-Z]{2})?$"
+          "pattern": "^(?i)[a-zA]{2}(-[a-z]{2})?$"
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply src/sarif-schema-2.1.0.json:2371]: Most likely. Though, I'd prefer to maintain parity with the underlying json schema. This doesn't change what matches the regex, so I'd prefer keeping it.

**Review by adityasharad — APPROVED** [review]

**Pull request merged**